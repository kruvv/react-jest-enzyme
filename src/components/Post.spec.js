import { shallow } from 'enzyme';
import React from 'react';
import Post from './Post';



const setUp = (props) => shallow(<Post {...props} />);
describe('should render component Post', ()=> {
    let component;
    beforeEach(() => {
        component = setUp();
    });
    it('should find className="post" in component Post', () => {
        const wrapper = component.find(".post");
        expect(wrapper.length).toBe(1);
        // Need for rendering component in terminal
        // console.log(component.debug());
    });
    it('should find tag <a> in component Post', () => {
        const wrapper = component.find("a");
        expect(wrapper.length).toBe(1);
    });
    it('should rendr created data', () => {
        const created_at = "02-09-2020";
        component = setUp({created_at});
        const date = component.find(".date");
        expect(date.text()).toBe(new Date(created_at).toLocaleDateString());
    });

    it("should correctly render Post", () => {
        expect(component).toMatchSnapshot();
    })

})
