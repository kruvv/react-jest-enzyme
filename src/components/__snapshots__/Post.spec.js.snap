// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render component Post should correctly render Post 1`] = `
<li
  className="post"
>
  <div
    className="discription"
  >
    <a
      className="title"
      href="#"
    >
      Here should be a title
    </a>
    <span
      className="text"
    >
      0 points
    </span>
    <span
      className="comments"
    >
      0 comments
    </span>
    <span
      className="date"
    >
      No Date
    </span>
    <span
      className="author"
    >
      Viktor
    </span>
  </div>
</li>
`;
